### PR TITLE
add 256/16m color support

### PR DESCRIPTION
**Pull request by Qix-**
In reference to #11.

Here it is. It adds two new properties, `color` and `bgColor`. Both properties each have three properties, `ansi16`, `ansi256`, and `ansi16m`. These properties each expose all of the [ANSI conversions available from `color-convert`](https://github.com/harthur/color-convert/blob/master/conversions.js#L69-L81).

Tests are inbound (waiting on high-level feedback).

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you not sort the props in package.json?

**Comment by sindresorhus** [comment]
// @jbnicolai @sunesimonsen @silverwind @kevva @arthurvr

File: package.json [hunk 1]
```diff
@@ -24,28 +24,32 @@
     "index.js"
   ],
   "keywords": [
+    "256",
     "ansi",
-    "styles",
+    "cli",
```
Comment (arthurvr) [inline package.json]: `cli` and `command-line` makes me think this package is more something like `chalk-cli`.

File: index.js [hunk 2]
```diff
@@ -1,5 +1,74 @@
 'use strict';
```
Comment (arthurvr) [inline index.js]: Remove this line.

File: package.json [hunk 3]
```diff
@@ -24,28 +24,32 @@
...
+    "styles",
+    "terminal",
+    "text",
+    "tty",
+    "xterm"
   ],
   "devDependencies": {
     "mocha": "*"
+  },
+  "dependencies": {
```
Comment (arthurvr) [inline package.json]: Move `dependencies` above `devDependencies`.

File: package.json [hunk 4]
```diff
@@ -5,14 +5,14 @@
...
+    "name": "Sindre Sorhus",
     "url": "sindresorhus.com"
   },
   "maintainers": [
-    "Sindre Sorhus <sindresorhus@gmail.com> (sindresorhus.com)",
+    "JD Ballard <i.am.qix@gmail.com> (github.com/qix-)",
     "Joshua Appelman <jappelman@xebia.com> (jbnicolai.com)",
-    "JD Ballard <i.am.qix@gmail.com> (github.com/qix-)"
+    "Sindre Sorhus <sindresorhus@gmail.com> (sindresorhus.com)"
   ],
```
Comment (arthurvr) [inline package.json]: Why did you change this? Also let's not change unrelated things in a random commmit ;)

File: package.json [hunk 5]
```diff
@@ -5,14 +5,14 @@
   "license": "MIT",
   "repository": "chalk/ansi-styles",
   "author": {
-    "name": "Sindre Sorhus",
     "email": "sindresorhus@gmail.com",
+    "name": "Sindre Sorhus",
```
Comment (arthurvr) [inline package.json]: Why did you change this? Also let's not change unrelated things in a random commmit ;)

File: index.js [hunk 6]
```diff
@@ -1,5 +1,74 @@
...
+var supportsColor = require('supports-color');
+
+supportsColor = supportsColor || {};
+supportsColor.level = supportsColor.level || 0;
+
+colorConvert.rgb.rgb = function rgb (rgb) {
+	return rgb;
+};
+
+function getConverter (color) {
```
Comment (arthurvr) [inline index.js]: Same thing with many named function expressions here, `function getConverter(color) {`. Notice the spacing.

File: index.js [hunk 7]
```diff
@@ -1,5 +1,74 @@
...
+	var has = function has (name) {
+		return color.hasOwnProperty(name);
+	}
+	switch (true) {
+		case has('r') && has('g') && has('b'): return 'rgb';
+		case has('h') && has('s') && has('l'): return 'hsl';
+		case has('h') && has('s') && has('v'): return 'hsv';
+		case has('h') && has('w') && has('b'): return 'hsv';
```
Comment (silverwind) [inline index.js]: hwb?

File: index.js [hunk 8]
```diff
@@ -1,5 +1,74 @@
 'use strict';
 
+var colorConvert = require('color-convert');
+var supportsColor = require('supports-color');
+
+supportsColor = supportsColor || {};
+supportsColor.level = supportsColor.level || 0;
+
+colorConvert.rgb.rgb = function rgb (rgb) {
```
Comment (arthurvr) [inline index.js]: Why are you overwriting this? Also no need for the extra naming:

```js
colorConvert.rgb.rgb = function (rgb) {
```

File: index.js [hunk 9]
```diff
@@ -1,5 +1,74 @@
...
+	return rgb;
+};
+
+function getConverter (color) {
+	var has = function has (name) {
+		return color.hasOwnProperty(name);
+	}
+	switch (true) {
```
Comment (arthurvr) [inline index.js]: I always prefer longer `if`-statements than this kind of tricks, but that's code style, at the end. @sindresorhus ?

File: index.js (hunk 7 above)
Comment (arthurvr) [inline index.js]: `hwb`?

Comment (arthurvr) [inline index.js]: Looks like @silverwind was quicker :p

File: index.js [hunk 10]
```diff
@@ -1,5 +1,74 @@
...
+		default: return null;
+	}
+}
+
+function getArrayFromKeyString(obj, str) {
+	if (str === 'keyword') {
+		return [obj['css']];
+	}
+
+	return str.split('').map(function(key) {
```
Comment (arthurvr) [inline index.js]: `function () {`

File: index.js [hunk 11]
```diff
@@ -1,5 +1,74 @@
...
+		return obj[key];
+	});
+}
+
+var levels = [
+	function level0 () {
```
Comment (arthurvr) [inline index.js]: The `level0`, `level1`, `level2`, ... naming is moot.

File: index.js [hunk 12]
```diff
@@ -1,5 +1,74 @@
...
+		var converter = getConverter(color);
+		if (!converter) {
+			return '';
+		}
+		var args = getArrayFromKeyString(color, converter);
+		var code = colorConvert[converter]['rgb'](args);
+		return '\u001b[' + (38 + offset) + ';2;' +
+			code[0] + ';' + code[1] + ';' + code[2] + 'm';
+	}
+];
```
Comment (arthurvr) [inline index.js]: The last three methods in this array seem, besides the last line, all pretty much exact. Let's try to keep it DRY?

**Comment by sindresorhus** [comment]
@bucaran

File: index.js (hunk 2 above)
Comment (Qix-) [inline index.js]: Why wouldn't we want a strict environment?

File: index.js (hunk 8 above)
Comment (Qix-) [inline index.js]: It doesn't exist to begin with.

File: index.js (hunk 6 above)
Comment (Qix-) [inline index.js]: That's how @sindresorhus does it.

File: index.js (hunk 7 above)
Comment (Qix-) [inline index.js]: Thanks, will push after the rest of these are resolved.

File: index.js (hunk 11 above)
Comment (Qix-) [inline index.js]: It's for debugging. I usually always add function names for stack traces. @sindresorhus?

File: index.js (hunk 12 above)
Comment (Qix-) [inline index.js]: Yep, I intend to refactor these. :+1:

File: package.json (hunk 5 above)
Comment (Qix-) [inline package.json]: Not sure why this changed. I didn't touch it; might have been NPM. I'll fix it.

File: package.json (hunk 4 above)
Comment (Qix-) [inline package.json]: Not sure why this changed. I didn't (consciously) touch it; might have been NPM. I'll fix it.

File: package.json (hunk 1 above)
Comment (Qix-) [inline package.json]: That's up to @sindresorhus. It appears that, despite my best efforts, package.json got manhandled by NPM. These just got sorted.

File: package.json (hunk 3 above)
Comment (Qix-) [inline package.json]: :+1:

**Comment by Qix-** [comment]
@sindresorhus it looks as if I did `npm --save` or `--save-dev` and didn't revert with the relevant changes. NPM manhandled it; I'll fix it.

File: index.js (hunk 9 above)
Comment (Qix-) [inline index.js]: I was trying to figure out a way to build these up dynamically and remove these horrid check to begin with, though properties aren't sorted anyway. This may be a [complete non-issue](https://github.com/chalk/chalk/issues/73#issuecomment-124826052)

File: index.js (hunk 2 above)
Comment (arthurvr) [inline index.js]: Take a good look, I didn't comment on the `'use strict';`, but on the blank line under that.

File: index.js (hunk 6 above)
Comment (arthurvr) [inline index.js]: It's not. `function name() {}` and `function () {}`.

File: index.js (hunk 8 above)
Comment (arthurvr) [inline index.js]: I'm not too fond on having to overwrite this. Might as well ask the author if his module can't just have a consistent API.

**Comment by sindresorhus** [comment]
> it looks as if I did npm --save or --save-dev and didn't revert with the relevant changes. NPM manhandled it; I'll fix it.

Weird, npm should only sort dependencies. That sounds like a npm bug.

**Comment by sindresorhus** [comment]
I'm a bit sceptical about using `supports-color` here.

**Comment by Qix-** [comment]
@sindresorhus nah, not a bug in NPM technically speaking. It uses `JSON.parse`, does its thing, and then dumps out the result back to `JSON.stringify`. The implementation of JSON doesn't honor original formatting. They'd have to rewrite their JSON parser in order to not interfere.

**Comment by Qix-** [comment]
@sindresorhus The alternative is to break out each ANSI format into its own subset of color conversion methods and let chalk take care of the support level. I'm in support of that; remembering this library should remain a reference to ANSI styles. Magic shouldn't be done here.

**Comment by sindresorhus** [comment]
> nah, not a bug in NPM technically speaking. It uses JSON.parse, does its thing, and then dumps out the result back to JSON.stringify. The implementation of JSON doesn't honor original formatting. They'd have to rewrite their JSON parser in order to not interfere.

In practise it does preserve it, except for keys starting with a number. It's npm that explicitly sort the dependencies objects.

**Comment by Qix-** [comment]
@sindresorhus [you're correct.](https://github.com/npm/npm/blob/master/lib/install/save.js#L104-L106)

File: index.js (hunk 6 above)
Comment (Qix-) [inline index.js]: Which line are you referring to?

File: index.js (hunk 8 above)
Comment (Qix-) [inline index.js]: Converting something to itself isn't a conversion. It wouldn't make sense to have it in there, realistically speaking. This conversion makes it so we have to perform one less check when building up which converter we want to use.

File: index.js (hunk 6 above)
Comment (arthurvr) [inline index.js]: Yes. Look at my first comment here ;)

Comment (Qix-) [inline index.js]: [I'm still confused](https://github.com/chalk/ansi-styles/pull/13#discussion_r35481365). The [line you're referring to](https://github.com/chalk/ansi-styles/pull/13#discussion-diff-35481365R13) has a space between `getConverter` and `()`. It's not a statement (i.e. not being assigned to something), so it's inarguably necessary to have a function name. What's the problem?

File: index.js (hunk 9 above)
Comment (Qix-) [inline index.js]: For the record, coffeescript does this.

File: index.js (hunk 6 above)
Comment (sunesimonsen) [inline index.js]: Shouldn't linting catch this?

Comment (arthurvr) [inline index.js]: @sunesimonsen We don't do code style linting (yet).

@Qix- It's as easy as `function getConverter (color) {` => `function getConverter(color) {`... We don't do the space between the function's name and the parenthesis.

Comment (sindresorhus) [inline index.js]: No worries about the code style though. I can easily fix it afterwards ;)

Comment (Qix-) [inline index.js]: @arthurvr Oh my. I could have sworn this is how we did things. >.> Will fix in next push.

File: index.js (hunk 11 above)
Comment (sindresorhus) [inline index.js]: Sure, but would probably look cleaner to define the functions in the top-level scope and just refer to them in the array.

**Comment by Qix-** [comment]
Latest push simplifies the lookups, and is forward-compatible with any additional conversions that may be added to `color-convert`. It doesn't use explicit checking for objects as per [chalk/chalk#73](https://github.com/chalk/chalk/issues/73#issuecomment-124833473).

File: index.js [hunk 13]
```diff
@@ -56,6 +79,38 @@ function assembleStyles () {
...
+		if (/2ansi$/.test(key)) {
+			styles.color.ansi256[conversion] = wrapAnsi256(fn, 0);
+			styles.bgColor.ansi256[conversion] = wrapAnsi256(fn, 10);
+			continue;
+		}
+		if (/2rgb$/.test(key)) {
+			styles.color.ansi16m[conversion] = wrapAnsi16m(fn, 0);
+			styles.bgColor.ansi16m[conversion] = wrapAnsi16m(fn, 10);
+			continue;
+		}
```
Comment (Qix-) [inline index.js]: inb4 @arthurvr suggests DRYing this out. I agree; what's the best way?

**Comment by Qix-** [comment]
@sindresorhus _*poke*_

**Comment by martinheidegger** [comment]
@Qix- I am not a member/owner here or anything but I think a documentation update and maybe tests would be important to merge this PR.

**Comment by Qix-** [comment]
Will write tests, I wanted to make sure the API is worthy first. @sindresorhus has been preoccupied with things lately, haha :+1:

**Comment by Qix-** [comment]
@sindresorhus poke x2

**Comment by Qix-** [comment]
Refresher: plan with this PR is to make sure the API is viable. If so, I'll add tests and docs and whatnot.

File: index.js [hunk 14]
```diff
@@ -56,6 +79,38 @@ function assembleStyles () {
...
+		ansi16: {},
+		ansi256: {},
+		ansi16m: {}
+	};
+
+	for (var key in colorConvert) {
+		var fn = colorConvert[key];
+		var conversion = key.replace(/^([^2]+).+$/, '$1');
+		if (/2ansi16$/.test(key)) {
+			styles.color.ansi16[conversion] = wrapAnsi16(fn, 0);
```
Comment (sindresorhus) [inline index.js]: `ansi16` and `ansi16m` is a bit too close? Any better name for either?

Comment (Qix-) [inline index.js]: Could either be

- `ansi16` -> `ansiBasic`
- `ansi16` -> `ansi`
- `ansi16m` -> `ansiTruecolor`

**Comment by Qix-** [comment]
Just a quick note about usage:

```javascript
styles.color.ansi256.rgb(r, g, b); // ansi 256 code from RGB
styles.color.ansi256.hsl(h, s, l); // ansi 256 code from HSL
```

All of the conversion supported can be found at MoOx/color-convert#18.

File: index.js (hunk 14 above)
Comment (sindresorhus) [inline index.js]: > ansi16 -> ansi

:+1:

File: index.js (hunk 13 above)
Comment (sindresorhus) [inline index.js]: Yeah, would like to see it simplified too, but not sure how or if it's possible, so happy to merge without if you can't find a way.

File: index.js (hunk 14 above)
Comment (Qix-) [inline index.js]: Agreed.

**Comment by Qix-** [comment]
Updated `color-convert` to 0.7.0, which includes hex support :+1:

**Comment by Qix-** [comment]
Tests are failing because ava is all messed up.

**Comment by sindresorhus** [comment]
I've fixed the tests, just rebase.

**Comment by sindresorhus** [comment]
The implementation looks good to me. Can you add some tests and docs?

**Comment by sindresorhus** [comment]
// @jbnicolai

**Comment by Qix-** [comment]
@sindresorhus Just noticed something. Not sure why I did it this way.

```javascript
styles.colors   .magenta;
styles.color    .ansi.rgb(1, 2, 3);

// and then, in the docs

ansi.color.ansi.rgb(12, 34, 56); // redundant
```

Should it be `styles.colors.ansi`? Should the docs change the imported variable from `ansi` to `styles` or `style`?

**Comment by Qix-** [comment]
Added docs and tests. Will change them according to any feedback regarding my previous comment.

File: index.js (hunk 13 above)
Comment (Qix-) [inline index.js]: It's possible but the tradeoff would be more code at the top. It'd probably even amount to more LOC in the end anyway. This is probably fine for now.

**Comment by sindresorhus** [comment]
> styles.colors   .magenta;
> styles.color    .ansi.rgb(1, 2, 3);

`color`, and make `bgColors` and `modifiers` singular too.

> Should it be styles.colors.ansi?

Yeah, let's keep it consistent.

> Should the docs change the imported variable from ansi to styles or style?

`style`

File: index.js (hunk 13 above)
Comment (sindresorhus) [inline index.js]: Ok

**Comment by Qix-** [comment]
Cool; one last thing: `style.green.close` for instance will always be `\u001b[39m`. Now that we have added `style.color.close` and `style.bgColor.close`, should we remove the individual close colors?

**Comment by sindresorhus** [comment]
> should we remove the individual close colors?

No, they're there to allow nesting of styles in `chalk`. 

From the chalk readme:

```js
// nest styles of the same type even (color, underline, background)
chalk.green(
    'I am a green line ' +
    chalk.blue.underline.bold('with a blue substring') +
    ' that becomes green again!'
);
```

**Comment by Qix-** [comment]
If for convenience, then I understand. Just wanted to point out that the close codes are the same for every color.

Otherwise, that's it for this PR. :+1:

File: readme.md [hunk 15]
```diff
@@ -69,17 +73,35 @@ Each style has an `open` and `close` property.
 
 By default you get a map of styles, but the styles are also available as groups. They are non-enumerable so they don't show up unless you access them explicitly. This makes it easier to expose only a subset in a higher-level module.
 
-- `ansi.modifiers`
-- `ansi.colors`
-- `ansi.bgColors`
+- `style.modifiers`
```
Comment (sindresorhus) [inline readme.md]: `modifiers` => `modifier`

Same with the actual code.

File: readme.md [hunk 16]
```diff
@@ -69,17 +73,35 @@ Each style has an `open` and `close` property.
...
 
 ###### Example
 
 ```js
-console.log(ansi.colors.green.open);
+console.log(style.color.green.open);
 ```
 
+## 256 / 16 million (TrueColor) support
+`ansi-styles` uses the [`color-convert`](/MoOx/color-convert) package to allow
```
Comment (sindresorhus) [inline readme.md]: Use the an absolute URL.

File: readme.md [hunk 17]
```diff
@@ -69,17 +73,35 @@ Each style has an `open` and `close` property.
...
+for converting between various colors and ANSI escapes, with support for 256
+and 16 million colors.
```
Comment (sindresorhus) [inline readme.md]: Don't hard-wrap.

File: readme.md [hunk 18]
```diff
@@ -69,17 +73,35 @@ Each style has an `open` and `close` property.
...
+
+To use these, call the associated conversion function with the intended output,
+e.g.:
+
+```javascript
```
Comment (sindresorhus) [inline readme.md]: `javascript` => `js`

File: readme.md [hunk 19]
```diff
@@ -17,11 +17,15 @@ $ npm install --save ansi-styles
...
-const ansi = require('ansi-styles');
+const style = require('ansi-styles');
 
-console.log(ansi.green.open + 'Hello world!' + ansi.green.close);
-```
+console.log(style.green.open + 'Hello world!' + ansi.green.close);
 
+// color conversion between 16/256/16 million colors
+console.log(style.bgColor.ansi.hsl(120, 80, 72) + 'Hello world!' + style.bgColor.close);
+console.log(style.color.ansi256.rgb(199, 20, 250) + 'Hello world!' + style.color.close);
```
Comment (sindresorhus) [inline readme.md]: The readme should say somewhere that this will be degraded to the closest color.

File: readme.md [hunk 20]
```diff
@@ -17,11 +17,15 @@ $ npm install --save ansi-styles
...
 ```js
-const ansi = require('ansi-styles');
+const style = require('ansi-styles');
 
-console.log(ansi.green.open + 'Hello world!' + ansi.green.close);
-```
+console.log(style.green.open + 'Hello world!' + ansi.green.close);
 
+// color conversion between 16/256/16 million colors
+console.log(style.bgColor.ansi.hsl(120, 80, 72) + 'Hello world!' + style.bgColor.close);
```
Comment (sindresorhus) [inline readme.md]: Do we really want to encourage this? It's only 8 colors to fall back to, so not very useful IMHO. Should we document it at all? If yes, what's the use-case for this?

File: readme.md [hunk 21]
```diff
@@ -69,17 +73,35 @@ Each style has an `open` and `close` property.
...
 
 
 ###### Example
 
 ```js
-console.log(ansi.colors.green.open);
+console.log(style.color.green.open);
 ```
 
+## 256 / 16 million (TrueColor) support
```
Comment (sindresorhus) [inline readme.md]: Maybe link up `256` and `16 million` to some pages where the user can read more about them?

Comment (sindresorhus) [inline readme.md]: Maybe also link up some compat table for 256 and 16m colors?

File: readme.md [hunk 22]
```diff
@@ -17,11 +17,15 @@ $ npm install --save ansi-styles
...
 
 ```js
-const ansi = require('ansi-styles');
+const style = require('ansi-styles');
 
-console.log(ansi.green.open + 'Hello world!' + ansi.green.close);
-```
+console.log(style.green.open + 'Hello world!' + ansi.green.close);
 
+// color conversion between 16/256/16 million colors
```
Comment (sindresorhus) [inline readme.md]: `16/256/16 million` looks wrong. It makes it read as it's either 16 million or 256 million or 16 million.

File: index.js [hunk 23]
```diff
@@ -57,6 +80,49 @@ function assembleStyles() {
...
+		rgb: wrapAnsi16m(rgb2rgb, 10)
+	};
+
+	for (var key in colorConvert) {
+		if (!colorConvert.hasOwnProperty(key)) {
+			continue;
+		}
+
+		var fn = colorConvert[key];
+		var conversion = key.replace(/^([^2]+).+$/, '$1');
```
Comment (sindresorhus) [inline index.js]: empty line below here

File: index.js [hunk 24]
```diff
@@ -57,6 +80,49 @@ function assembleStyles() {
...
+		if (/2ansi16$/.test(key)) {
+			styles.color.ansi[conversion] = wrapAnsi16(fn, 0);
+			styles.bgColor.ansi[conversion] = wrapAnsi16(fn, 10);
+			continue;
+		}
```
Comment (sindresorhus) [inline index.js]: empty line below here

File: index.js [hunk 25]
```diff
@@ -57,6 +80,49 @@ function assembleStyles() {
...
+		if (/2ansi$/.test(key)) {
+			styles.color.ansi256[conversion] = wrapAnsi256(fn, 0);
+			styles.bgColor.ansi256[conversion] = wrapAnsi256(fn, 10);
+			continue;
+		}
```
Comment (sindresorhus) [inline index.js]: empty line below here

File: index.js (hunk 23 above)
Comment (sindresorhus) [inline index.js]: wish we didn't have to use all this regex on the `color-convert` methods.

File: readme.md [hunk 26]
```diff
@@ -17,11 +17,15 @@ $ npm install --save ansi-styles
 ## Usage
 
 ```js
-const ansi = require('ansi-styles');
+const style = require('ansi-styles');
 
-console.log(ansi.green.open + 'Hello world!' + ansi.green.close);
-```
+console.log(style.green.open + 'Hello world!' + ansi.green.close);
```
Comment (sindresorhus) [inline readme.md]: `ansi.green.close` => `style.green.close`

File: readme.md (hunk 18 above)
Comment (Qix-) [inline readme.md]: I'll change it for consistency but technically both work.

File: readme.md (hunk 20 above)
Comment (Qix-) [inline readme.md]: Because chalk will need to know this for degrading colors down as well in the event `supports-color` returns `1`.

File: readme.md (hunk 22 above)
Comment (Qix-) [inline readme.md]: How about `16/256/truecolor`?

File: index.js (hunk 23 above)
Comment (Qix-) [inline index.js]: Me neither. The hash it has is just `[from][to]`; getting a list of `[to]` by their names would cause two loops. I guess we could do it that was but this was the most concise.

File: index.js (hunk 25 above)
Comment (Qix-) [inline index.js]: Shouldn't XO be catching these?

File: readme.md (hunk 20 above)
Comment (sindresorhus) [inline readme.md]: Ok, maybe just add a note that instead of using it directly (not through chalk) it's better to use one of the 8 colors directly.

File: readme.md (hunk 22 above)
Comment (sindresorhus) [inline readme.md]: :+1:

File: index.js (hunk 23 above)
Comment (sindresorhus) [inline index.js]: > [from][to]

I don't get it. Can't you convert in both directions? So shouldn't it have both?

I would be fine with two loops if it got rid of the regex stuff.

File: index.js (hunk 25 above)
Comment (sindresorhus) [inline index.js]: @Qix- Ideally, yes, but ESLint has no such rule :/ Might be worth proposing it.

File: readme.md (hunk 18 above)
Comment (sindresorhus) [inline readme.md]: Yup, but you know me, I'm a consistency freak.

File: index.js (hunk 25 above)
Comment (Qix-) [inline index.js]: Could have sworn they did.. That's a common linting rule in other languages.

File: index.js (hunk 23 above)
Comment (Qix-) [inline index.js]: You can in theory, but the API may not support it.

Further, `for (var k in convert['ansi256'])` would return a bunch of keys to a bunch of functions _from_ 256 colors, which doesn't really help us. We'd have to cache them and then reverse-look them up too, which is much messier than two loops.

File: readme.md (hunk 21 above)
Comment (Qix-) [inline readme.md]: Can-Do&trade;

File: readme.md (hunk 20 above)
Comment (Qix-) [inline readme.md]: :+1:

File: readme.md (hunk 18 above)
Comment (Qix-) [inline readme.md]: consistency

File: readme.md (hunk 17 above)
Comment (Qix-) [inline readme.md]: Sorry, ViM habit.

File: readme.md (hunk 16 above)
Comment (Qix-) [inline readme.md]: :+1:

File: index.js (hunk 25 above)
Comment (sindresorhus) [inline index.js]: @Qix- I thought so too, but no.

http://eslint.org/docs/rules/#stylistic-issues

File: index.js (hunk 23 above)
Comment (sindresorhus) [inline index.js]: Alright, two loops it is then.

File: index.js (hunk 25 above)
Comment (Qix-) [inline index.js]: ._. disappointing.

Comment (sindresorhus) [inline index.js]: @Qix- https://github.com/eslint/eslint/issues/new ;)

Comment (Qix-) [inline index.js]: Ugh I'll do it tomorrow. Their rule request requirements require a requiem to [rhamnusia](http://www.godchecker.com/pantheon/greek-mythology.php?deity=NEMESIS).

File: readme.md (hunk 20 above)
Comment (Qix-) [inline readme.md]: On second look at this, what was the problem here?

**Comment by Qix-** [comment]
That should do it.

**Comment by sindresorhus** [comment]
Landed, finally. Woop woop.

Thanks @Qix- for perseverating through this loooooong process.

thans

**Comment by sindresorhus** [comment]
Now we just need to resolve https://github.com/chalk/chalk/issues/73.

**Comment by silverwind** [comment]
Awesome! :beers:

**Comment by Qix-** [comment]
Woo!

**Comment by Qix-** [comment]
What happened in this PR:

- `.colors` / `.bgColors` / `.modifiers` -> `.color` / `.bgColor` / `.modifier` (plural -> singular)
- `.color.ansi.XXX()` - returns a 16 color code given a [`color-convert`](/Qix-/color-convert) conversion (e.g. `.color.ansi.rgb(255, 160, 50)`)
- `.color.ansi256.XXX()` - returns a 256 color code
- `.color.ansi16m.XXX()` - returns a truecolor (16 million) color code
